src/metrics.py

```diff
@@ -0,0 +1,32 @@
+#  Copyright 2023 Canonical Ltd.
+#  See LICENSE file for licensing details.
+
+"""Models and functions for the metric events."""
+
+from pydantic import BaseModel, conint
+
+
+class Event(BaseModel):
+    """Base class for metric events."""
+
+    timestamp: conint(ge=0)
+    """The UNIX timestamp of the event."""
+
+
+class RunnerInstalled(Event):
+    """Metric event for when a runner is installed."""
+
+    name: str = "runner_installed"
+    """The name of the event."""
```

Comment:
It would be good to describe attributes using something other than the name of it, e.g., Short description of the event, what a flavour is meant to be, possible example values or something like that. This means that the description adds extra information

Reply:
Updated. Removed the name attribute, as the name can also be retrieved by the class name. This would be used before sending the final log to Prometheus.